File: C++/Unit-4 moodle_practical/qu13.cpp
// Write an overloaded template function area that calculates area of different shapes
#include<iostream>
using namespace std;
class over_load
{
    public:
    template<class T>
    void func(T r)
    {
        T pi=3.14;

        T ans=pi*r*r;

        cout<<"Area of circle is : "<<ans<<endl;;
    }
    template<class Y>
    void func(Y l,Y b)
    {
        Y ans=l*b;

        cout<<"Area of square is : "<<ans<<endl;
    }
};

int main()
{
    over_load obj;
    over_load obj1;

    int r;
    cout<<"Enter r for area of circle : "<<endl;
    cin>>r;
    obj.func(r);

    int l,b;
    cout<<"Enter l and b for area of rectangle : "<<endl;
    cin>>l>>b;
    obj.func(l,b);

}

File: sem 4/JAVA/UNIT_3/Qu13.java
class Rect
{
    int length;
    int breath;

    Rect(int length,int breath)
    {
        this.length=length;
        this.breath=breath;
    }

    void Area()
    {
        int ans=length*breath;
        System.out.println("Area is : "+ans);
    }

    void Perimeter()
    {
        int result=2*(length+breath);
        System.out.println("Perimeter is : "+result);
    }
}

class square extends Rect
{
    square(int side)
    {
        super(side,side);
    }
}

class Qu13
{
    public static void main(String[] args) 
    {
        Rect obj=new Rect(5, 5);

        obj.Area();//(5*5)
        obj.Perimeter();//2(5+5)

        square obj1=new square(9);

        obj1.Area();//(9*9)
        obj1.Perimeter();//2(9+9)
    }
}
